tests/utils/addParticipant.test.ts

```diff
@@ -0,0 +1,93 @@
+import { addParticipant } from "../../src/utils";
+import client from '../../src/twilioClient';
+
+import { ParticipantInstance, ParticipantListInstanceCreateOptions } from "twilio/lib/rest/conversations/v1/conversation/participant";
+
+
+jest.mock('../../src/twilioClient')
+let mockedClient = jest.mocked(client, true)
+
+const mockParticipant: Partial<ParticipantListInstanceCreateOptions> = {
+  identity: "+1234",
+}
+
+describe('addParticipant util', () => {
+  beforeEach(() => {
+    jest.resetAllMocks()
+  })
+
+   
+  it('it adds participant to conversation', async () => {
+    let createSpy = jest.fn((options) => { return mockParticipant })
+    const conversationsSpy = jest.fn((options) => {
+      return {
+        participants: { create: createSpy }
+      }
+    });
+
+    mockedClient['conversations'] = {
+        conversations: conversationsSpy
+    } as any
+
+
+    const result = await addParticipant("myConversationSid", mockParticipant );
+    expect(conversationsSpy).toBeCalledWith("myConversationSid");
+    expect(createSpy).toBeCalledWith(mockParticipant);
+    expect(result).not.toBeNull();
+  })
+
+  it('calls quit if error is not a 429 retry', async () => {
+    let createSpy = jest.fn((options) => { throw new Error('Twilio Problem') })
+    const conversationsSpy = jest.fn((options) => {
+      return {
+        participants: { create: createSpy }
+      }
+    });
+
+    mockedClient['conversations'] = {
+        conversations: conversationsSpy
+    } as any
+
+    const consoleSpy = jest.spyOn(console, 'log');
+
+    try {
+      await addParticipant("myConversationSid", mockParticipant );
+    } catch (e) {
+      expect(consoleSpy).toHaveBeenCalledWith('Quit without retry');
```

Comment:
As far as I understand this verification will be called only if exception occurs, if not test would just pass as is. I would suggest moving it outside of catch block.

Reply:
yeah this is strange, don't think this is right.